tests/suites/test_suite_psa_crypto_se_driver_hal_mocks.function

```diff
@@ -108,6 +117,23 @@ psa_status_t mock_export( psa_drv_se_context_t *context,
     return( mock_export_data.return_value );
 }
 
+psa_status_t mock_export_public( psa_drv_se_context_t *context,
+                                 psa_key_slot_number_t slot_number,
+                                 uint8_t *p_data,
+                                 size_t data_size,
+                                 size_t *p_data_length )
+{
+    (void) context;
+    (void) p_data;
+    (void) p_data_length;
+
+    mock_export_public_data.called++;
+    mock_export_public_data.slot_number = slot_number;
+    mock_export_public_data.data_size = data_size;
```

Comment:
Are we going to check anything about what the core gave us for slot_number or data_size? I don't see the point in saving these unless we look at their values during our test, to ensure the core passed on the information from the top level to the driver level correctly.

Reply:
The idea behind saving all of the parameters (apart from pointers) passed to the mocks was that it's irrelevant for them what the tests check and once the mocks have been created, no real changes to them need to be introduced. If anyone wants to extend the tests and check a different parameter - only a test has to be added/updated.

Reply:
Another point in favor of saving the values by default is that we'll get unused-parameter errors if we don't do something with them.

Reply:
OK, please make sure we are consistently saving all the values (other than pointers).